var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -274,10 +274,11 @@ def cmake_args(self):
     #
     # export SPACK_USER_CACHE_PATH=/tmp/spack
     # export SPACK_DISABLE_LOCAL_CONFIG=true
-    def test(self):
+
+    def test_develop(self):
+    """Various tests for @develop"""
         if not self.spec.satisfies("@develop") or not os.path.isdir(self.prefix.bin):
-            tty.info("Skipping: checks not installed in bin for v{0}".format(self.version))
-            return
+            raise SkipTest("Skipping: checks not installed in bin for v{0}".format(self.version))
 
```

Comment:
Instead of `test_develop` I would create two test methods: `test_N1pMDsEx1` and `test_N1pMdsEx1Raja`.  Each would iterate over the options in the list.